docs/topics/tour/kotlin-tour-intermediate-functions.md

```diff
@@ -0,0 +1,935 @@
+member functions in the `main()` function.
+
+Although this example is compact, in the real world it can be a while after the creation of your class instance before you
+configure it and use its member functions. However, if you use the `.apply()` scope function you can create, configure and
+use member functions on your class instance all in the same place in your code:
+
+```kotlin
+class Client() {
+  var token: String? = null
+  fun connect() = println("connected!")
+  fun authenticate() = println("authenticated!")
+  fun getData(): String = "Mock data"
+}
+//sampleStart
+val client = Client().apply {
+  token = "asdf"
+  connect()
+  authenticate()
+}
+
+fun main() {
+  client.getData()
+  // connected!
+  // authenticated!
+}
+//sampleEnd
+```
+{kotlin-runnable="true" id="kotlin-tour-scope-function-apply-after"}
+
+The example:
+
+* Creates `client` as an instance of the `Client` class.
+* Uses the `.apply()` scope function on the `client` instance.
+* Creates a temporary scope within the `.apply()` scope function so that you don't have to explicitly refer to the `client` instance when accessing its properties or functions.
+* Passes a lambda expression to the `.apply()` scope function that updates the `token` property and calls the `connect()` and `authenticate()` functions.
+* Calls the `getData()` member function on the `client` instance in the `main()` function.
+
+As you can see, this strategy is convenient when you are working with large pieces of code.
+
+### Run
+
+Similar to `.apply()` you can use the `.run()` scope function to initialize an object, but it's better to use `.run()` 
+to initialize an object at a specific moment in your code **and** immediately compute a result.
+
+Let's continue the previous example for the `.apply()` function but this time you want the `connect()` and
+`authenticate()` functions to be grouped so that they are called on every request.
+
+For example:
+
+```kotlin
+class Client() {
+    var token: String? = null
+    fun connect() = println("connected!")
+    fun authenticate() = println("authenticated!")
+    fun getData(): String = "Mock data"
+}
+
+//sampleStart
+val client: Client = Client().apply {
+    token = "asdf"
+}
+
+fun main() {
+    val result: String = client.run {
+        connect()
+        // connected!
+        authenticate()
+        // authenticated!
+        getData()
+    }
+}
+//sampleEnd
+```
+{kotlin-runnable="true" id="kotlin-tour-scope-function-run"}
+
+The example:
+
+* Creates `client` as an instance of the `Client` class.
+* Uses the `.apply()` scope function on the `client` instance.
+* Creates a temporary scope within the `.apply()` scope function so that you don't have to explicitly refer to the `client` instance when accessing its properties or functions.
+* Passes a lambda expression to the `.apply()` scope function that updates the `token` property.
+
+|--|--|
```

Comment:
I was trying to make it clear that the comments are related to before/after the main function and break it up so it's not just a wall of text. Do you think it would be better to combine the two lists?

Reply:
No, l'd vote for two separate lists with intro sentences for both.
because a single list seems too big, but the blank space between two lists seems like a bug